tesseract_command_language/include/tesseract_command_language/profile_dictionary.h

```diff
@@ -191,11 +159,50 @@ class ProfileDictionary
   std::shared_ptr<const ProfileType> getProfile(const std::string& ns, const std::string& profile_name) const
   {
     std::shared_lock lock(mutex_);
-    const auto& it = profiles_.at(ns);
-    const auto& it2 = it.at(std::type_index(typeid(ProfileType)));
+
+    const std::unordered_map<std::type_index, std::any>* it = nullptr;
+    try
```

Comment:
I would avoid a catch here and just check if it exists and if not throw exception.

Reply:
Both catches here simply rethrow the exception with more, relevant information. We could do a check with `it->find(...)` and then throw an exception if the key does not exist, but that's functionally the same as what is here already. In either case, a terminating exception gets thrown and (almost) nothing downstream in the planning framework catches it, so it doesn't seem like the catch is really that expensive

Reply:
Yea, it seem this is something that should not happen in normal operation.